src/connectivity/providers/linux.rs

```diff
@@ -51,4 +53,25 @@ impl Connectivity for WiFi {
             .as_ref()
             .contains("disconnect"))
     }
+
+    fn speed(&self) -> Result<String, WifiConnectionError> {
+        let nmcli_speed = Command::new("nmcli")
+            .arg("dev")
+            .arg("wifi")
+            .arg("list")
+            .stdout(Stdio::piped())
+            .spawn()
+            .expect("nothing");
+        let piped_awk = Command::new("awk")
+            .arg(r"/\*/{if (NR!=1) {print $6}}")
+            .stdin(Stdio::from(nmcli_speed.stdout.unwrap())) // Pipe through.
+            .stdout(Stdio::piped())
+            .spawn()
+            .unwrap();
```

Comment:
This should be parsed manually. We'll want to reduce as much dependency as possible on external programs. 

In the future, I plan to remove dependency on `nmcli` for a similar reason.

Reply:
Also, when parsing manually, could you add test cases to verify we're parsing the output of `nmcli` correctly?